### PR TITLE
musa: add docker image support

### PR DESCRIPTION
**Pull request by yeahdongcn**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

This PR introduces Dockerfiles for `musa`, allowing the build of 3 Docker images: `local/llama.cpp:full-musa`, `local/llama.cpp:light-musa`, and `local/llama.cpp:server-musa`.

## Testing done
- [x] Successfully built `musa` images.
- [x] Verified each Docker image runs with `mt-container-toolkit` and `Llama 3.2 1B Instruct`.

### Logs
[server-musa.log](https://github.com/user-attachments/files/17182491/server-musa.log)
[light-musa.log](https://github.com/user-attachments/files/17182492/light-musa.log)
[full-musa.log](https://github.com/user-attachments/files/17182493/full-musa.log)

### CODE REVIEW

File: docs/docker.md [hunk 1]
```diff
@@ -19,8 +19,11 @@ Additionally, there the following images, similar to the above:
 - `ghcr.io/ggerganov/llama.cpp:full-rocm`: Same as `full` but compiled with ROCm support. (platforms: `linux/amd64`, `linux/arm64`)
 - `ghcr.io/ggerganov/llama.cpp:light-rocm`: Same as `light` but compiled with ROCm support. (platforms: `linux/amd64`, `linux/arm64`)
 - `ghcr.io/ggerganov/llama.cpp:server-rocm`: Same as `server` but compiled with ROCm support. (platforms: `linux/amd64`, `linux/arm64`)
+- `ghcr.io/ggerganov/llama.cpp:full-musa`: Same as `full` but compiled with MUSA support. (platforms: `linux/amd64`)
+- `ghcr.io/ggerganov/llama.cpp:light-musa`: Same as `light` but compiled with MUSA support. (platforms: `linux/amd64`)
+- `ghcr.io/ggerganov/llama.cpp:server-musa`: Same as `server` but compiled with MUSA support. (platforms: `linux/amd64`)
```
Comment (slaren) [inline docs/docker.md:24]: Note that for the images to be uploaded to ghcr.io they need to be added to `docker.yml`.

Comment (yeahdongcn) [reply docs/docker.md:24]: Thanks for the hints! I plan to add them in a separate PR, as the final image of `musa:3.1.0-devel/runtime` will be available after our National Day holiday.

**Review by slaren — APPROVED** [review]

File: docs/docker.md (hunk 1 above)
Comment (yeahdongcn) [reply docs/docker.md:24]: Please see the draft PR: https://github.com/ggerganov/llama.cpp/pull/9691

Comment (slaren) [reply docs/docker.md:24]: I can merge this now if you want, but until this is added to the ci, these images are not going to be available on ghcr.io for the users trying to follow instructions from this document.

Comment (yeahdongcn) [reply docs/docker.md:24]: I understand your concern. It seems combining the two PRs might be a better approach. Let's keep this one open until the base images are ready and published.

I’ll convert it to a draft. Let’s revisit once the base images are ready.

Comment (yeahdongcn) [reply docs/docker.md:24]: > Please see the draft PR: #9691

Closed.

**Comment by yeahdongcn** [comment]
Everything is set now. @slaren, could you please take a look? Thanks!

**Review by slaren — APPROVED** [review]

**Pull request merged**